wp-includes/sqlite/class-wp-sqlite-db.php

```diff
@@ -249,12 +287,12 @@ public function query( $query ) {
 		}
 
 		$this->last_error = $this->dbh->get_error_message();
-		if ( $this->last_error && ( ! defined( 'WP_INSTALLING' ) || ! WP_INSTALLING ) ) {
+		if ( $this->last_error ) { // && ( ! defined( 'WP_INSTALLING' ) || ! WP_INSTALLING ) ) {
 			$this->print_error( $this->last_error );
 			return false;
 		}
 
-		if ( preg_match( '/^\\s*(create|alter|truncate|drop|optimize)\\s*/i', $query ) ) {
+		if ( preg_match( '/^\\s*(set|create|alter|truncate|drop|optimize)\\s*/i', $query ) ) {
```

Comment:
Ideally these checks would be handled in the engine class – it already sets all this data based on a query type.